src/Control/Controller.php

```diff
@@ -300,6 +301,38 @@ protected function handleAction($request, $action)
         }
 
         $result = $this->getViewer($action)->process($this);
+        
```

Comment:
I think this kind of process that analyse the HTTP Response and alters it is better handle as a Middleware.